### PR TITLE
Adding deprecation message to each package.

### PR DESCRIPTION
**Pull request by marstr**
This work is with regards to #868.

---

Thanks you for your contribution to the Azure-SDK-for-Go! We will triage and review it as quickly as we can.

As part of your submission, please make sure that you can make the following assertions:

 - [x] I'm not making changes to Auto-Generated files which will just get erased next time there's a release.
   - If that's what you want to do, consider making a contribution here: https://github.com/Azure/autorest.go
 - [x] I've tested my changes, adding unit tests where applicable.
 - [x] I've added Apache 2.0 Headers to the top of any new source files.
 - [x] I'm submitting this PR to the `dev` branch, or I'm fixing a bug that warrants its own release and I'm targeting the `master` branch.
 - [x] If I'm targeting the `master` branch, I've also added a note to [CHANGELOG.md](https://github.com/Azure/azure-sdk-for-go/blob/master/README.md).
 - [x] I've mentioned any relevant open issues in this PR, making clear the context for the contribution.

### CODE REVIEW

**Comment by mcardosos** [comment]
Is the plan for this to be released with v12?

**Comment by vladbarosan** [comment]
> Is the plan for this to be released with v12?

I would assume we want to mark them as deprecated ASAP once the other ones are active. Otherwise its just confusing.

File: services/resources/deployment.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package deployment
```
Comment (vladbarosan) [inline services/resources/deployment.go]: from what @marstr mentioned this was a manual maintained package. Maybe create an issue to investigate whats up with it.

Comment (marstr) [reply services/resources/deployment.go]: I create #877 to track this.

**Pull request merged**